src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Dependencies/IDependencyCollection.cs

```diff
@@ -1,18 +1,40 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.DotNet.UpgradeAssistant.Dependencies
 {
     public interface IDependencyCollection<T> : IEnumerable<T>
     {
-        bool Add(T item, BuildBreakRisk risk = BuildBreakRisk.None);
+        bool Add(T item, OperationDetails details);
```

Comment:
So this will break current implements (ie loose dependency). Can we phase in the new API?

what about 

```csharp
[Obsolete]
[EditorBrowseable(EditorBrowseableState.None)]
bool Add(T item, BuildBreakRisk risk = BuildBreakRisk.None);

bool Add(T item, OperationDetails detail);

...etc
```